packages/vm/package.json

```diff
@@ -52,7 +55,7 @@
     "test:state:allForks": "echo 'Chainstart Homestead dao TangerineWhistle SpuriousDragon Byzantium Constantinople Petersburg Istanbul MuirGlacier Berlin London ByzantiumToConstantinopleFixAt5 EIP158ToByzantiumAt5 FrontierToHomesteadAt5 HomesteadToDaoAt5 HomesteadToEIP150At5 BerlinToLondonAt5 Cancun' | xargs -n1 | xargs -I v1 npm run test:state -- --fork=v1 --verify-test-amount-alltests",
     "test:state:selectedForks": "echo 'Homestead TangerineWhistle SpuriousDragon Petersburg Berlin London Cancun' | xargs -n1 | xargs -I v1 npm run test:state -- --fork=v1 --verify-test-amount-alltests",
     "test:state:slow": "npm run test:state -- --runSkipped=slow",
-    "tester": "tsx ./test/tester --stack-size=1500",
+    "tester": "tsx --conditions=typescript ./test/tester --stack-size=1500",
```

Comment:
Not a super strong opinion but I think we should keep the behavior the same (use the built packages instead of reading source directly) and instead using the environment flag to activate it.

Reply:
There isn't, I think my main point would be to keep it all consistent. So then all the `test:node` should also use the typescript condition so they will take the source code of other monorepo packages instead of using the "old" compiled versions.

Small other thought, we can now likely speed up the CI because the "build" step is not necessary anymore for tests (?). (However we need a build CI job to see if the builds themselves succeed)

Reply:
Okay, I think I see the disconnect here.  So, vitest doesn't respect the `--conditions=typescript` flag in the CLI.  That's a `node` executable flag and vitest just ignores it.  That's why I specified it in the new monorepo level `vitest.config.mts` configuration. But the tester doesn't use vitest but is actually running with `tsx` (which itself is node + an esbuild prestep to strip out types) and `tsx` does respect node CLI flags.

Does that make sense?

Reply:
[Here](https://github.com/vitest-dev/vitest/issues/6992) is where I learned about vitest not respecting flags.

Reply:
Right